processing/src/main/java/org/apache/druid/query/aggregation/first/DoubleFirstAggregatorFactory.java

```diff
@@ -149,7 +150,7 @@ public VectorAggregator factorizeVector(
           timeColumn);
       return new DoubleFirstVectorAggregator(timeSelector, valueSelector);
     }
-    return NumericNilVectorAggregator.doubleNilVectorAggregator();
+    return NumericNilVectorAggregator.of(new SerializablePair<>(0L, NullHandling.defaultDoubleValue()));
```

Comment:
this could be saved as a static variable instead of making a new one each time since its not going to change over the lifetime of the service, same for all the other impls